test/env.jl

```diff
@@ -30,3 +30,36 @@ key = randstring(25)
 @test !haskey(ENV,key)
 @test_throws KeyError ENV[key]
 @test get(ENV,key,"default") == "default"
+
+# Test for #17956
+@test length(ENV) > 1
+k1, k2 = "__test__", "__test1__"
+withenv(k1=>k1, k2=>k2) do
+    b_k1, b_k2 = false, false
+    for (k, v) in ENV
+        if k==k1
+            b_k1=true
+        elseif k==k2
+            b_k2=true
+        end
+    end
+    @test b_k1 && b_k2
+    io = IOBuffer()
+    show(io, ENV)
+    s = takebuf_string(io)
+    @test contains(s, k1)
```

Comment:
should we be even more specific and check for `"$k1 => $k1"` (maybe with some whitespace padding as needed)?